certbot/tests/main_test.py

```diff
@@ -1795,5 +1716,115 @@ def test_install_enhancement_no_certname(self, mock_inst, _rec):
                           self.config, plugins)
 
 
+class UpdateAccountTest(test_util.ConfigTestCase):
+    """Tests for certbot._internal.main.update_account"""
+
+    def setUp(self):
+        self.patches = {
+            'account': mock.patch('certbot._internal.main.account'),
+            'atexit': mock.patch('certbot.util.atexit'),
+            'client': mock.patch('certbot._internal.main.client'),
+            'determine_account': mock.patch('certbot._internal.main._determine_account'),
+            'notify': mock.patch('certbot._internal.main.display_util.notify'),
+            'prepare_sub': mock.patch('certbot._internal.eff.prepare_subscription'),
+            'util': test_util.patch_get_utility()
+        }
+        self.mocks = { k: self.patches[k].start() for k in self.patches }
+
+        return super(UpdateAccountTest, self).setUp()
+
+    def tearDown(self):
+        for patch in self.patches.values():
+            patch.stop()
```

Comment:
If we're going to use this approach to patching, I think we should use `addCleanup` to call `stop` rather than doing it in `tearDown`. See https://github.com/certbot/certbot/issues/3874.